### PR TITLE
HPCC-23734 EclWatch React/Dojo Wrapper

### PR DESCRIPTION
**Pull request by GordonSmith**
React Component for Dojo Widgets.

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
Simple Usage:
```javascript
export const Test: React.FunctionComponent<{}> = ({ }) => {
    const divStyle = {
        width: "100%",
        height: "320px"
    };
    return <div>
        <h1>WU Queries</h1>
        <div style={divStyle}>
            <DojoComponent widget="GraphsWUWidget" params={{ Wuid: "W20200317-193149" }} />
        </div>
    </div>;
}

```

**Comment by GordonSmith** [comment]
And you can see a demo React Frame by adding the following to your launch config:
```
        {
            "name": "React Frame",
            "url": "http://localhost:8080/esp/files/index.html",
            "webRoot": "${workspaceRoot}",
            "type": "chrome",
            "request": "launch"
        },
```
And building as normal...

File: esp/src/index.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<!DOCTYPE html>
+<html lang="en">
+
```
Comment (miguelvazq) [inline esp/src/index.html]: Minor: extra space

File: esp/src/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    basePath: baseHost + "/esp/files",
+    fullPath: location.origin + "/esp/files"
+};
+
+const store = globalKeyValStore();
+store.set("", "", false, "HPCCApps", "ECLWatch");
+initSession();
+
+const theme = createMuiTheme({
+    palette: {
```
Comment (miguelvazq) [inline esp/src/src/index.tsx]: What theme are you adding here? Is this a custom theme just for this component?

**Review by miguelvazq — COMMENTED** [review]
One question, one minor.

File: esp/src/src/index.tsx (hunk 2 above)
Comment (GordonSmith) [reply esp/src/src/index.tsx]: No idea - its just cut and paste from the material-ui Getting Started App.  This PR is about the DojoComponent, not about the index.html/index.tsx as such.

File: esp/src/index.html (hunk 1 above)
Comment (miguelvazq) [reply esp/src/index.html]: Lets make sure we do not include the index.html as part of the package

**Comment by GordonSmith** [comment]
@miguelvazq made change to prevent the "index" getting included in a production build.

**Review by miguelvazq — APPROVED** [review]

**Review by kunalaswani — APPROVED** [review]

**Pull request merged**